### DISCUSSION TITLE
Mock API backend by other language like C# or Python?

### DISCUSSION THREAD

**Discussion by xiazeyu** [Difegue/LANraragi#692] [Ideas]
This is based on these following problems:

- My friend want to manage his manga, but he doesn’t have a NAS. He want to put everything into a USB flash drive or portable hard disk.
- Perl is painful on Windows. There’s Strawberry Perl and ActivateState Perl, but I don’t know how to use (my bad)
- Current WSL solution is even more painful for me. It’s just not native, not support previous version of Windows(that’s fair), and sometimes there will be weird problems like path, installation or something else.
- Current UTF-8 support is still confusing

I don’t know why this project is originally written in Perl. And I have little knowledge about Perl.

So I am thinking of mock current API in Python or C#. It can be difficult to maintain both, and it needs tons of efforts(but only implement simple functions like read may be fine).

Please tell me your thoughts about this.

**Comment by Difegue** [comment]
I wanted to reply to this as soon as I saw it but forgot about it, so my apologies 🙇  

Frankly speaking, I think this'd be a waste of time.
There's been at least one attempt at recreating the API that I know of -- It was used by some rogue fakku mirror website which you could then only access through external clients like LRReader or Tachiyomi.  

![image](https://user-images.githubusercontent.com/8237712/188742921-434ee30f-ecaf-43af-8de2-542f00447003.png)  

The API being an incomplete recreation meant that half the functionalities of the clients wouldn't work at all and lead to crashes 'n stuff. [It wasn't very good.](https://github.com/Guerra24/LRReader/releases/tag/v1.7.2.0)  
Now of course I can't tell you what to do, so if you think the API is good enough to copy and make your own, you can certainly go ahead!  
But I certainly **won't** ever take on the burden of merging/maintaining a second version of the API written in a different language. ¯\\\_(ツ)_/¯  

Most of your problems sadly are fundamental (this is software designed for NASes to begin with), and I think you might have better luck simply using something else than LRR at this point! 🙃  
  
I'll agree that Windows support is sadly inferior at the moment: Building on top of WSL worked fine at the time of WSL1 but has steadily gotten worse as Microsoft focused on WSL2, whose virtual machine underpinnings make LRR seem much less "native" that it used to be.
